src/main/java/org/glite/authz/pep/pip/provider/PolicyNamesPIPIniConfigurationParser.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) Members of the EGEE Collaboration. 2006-2010.
+ * See http://www.eu-egee.org/partners/ for details on the copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors:
+ * 2016-
+ *    Mischa Salle <msalle@nikhef.nl>
+ *    Rens Visser <rensv@nikhef.nl>
+ *    NIKHEF Amsterdam, the Netherlands
+ *    <grid-mw-security@nikhef.nl>
+ */
+
+package org.glite.authz.pep.pip.provider;
+
+import org.glite.authz.common.config.AbstractConfigurationBuilder;
+import org.glite.authz.common.config.ConfigurationException;
+import org.glite.authz.common.config.IniSectionConfigurationParser;
+import org.glite.authz.pep.pip.PolicyInformationPoint;
+
+import java.io.IOException;
+
+import org.ini4j.Ini;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The configuration parser for the PolicyNamesPIP.
+ * @author Mischa Salle, Rens Visser
+ */
+public class PolicyNamesPIPIniConfigurationParser
+    implements IniSectionConfigurationParser<PolicyInformationPoint>
+{
+    /** Class logger. */
+    private Logger log = LoggerFactory.getLogger( PolicyNamesPIPIniConfigurationParser.class);
```

Comment:
I would have the logger static and final for this PIP, since we don't plan to have multiple instances in a PIP chain, do we, and even in that case, do we want to have logging specific to the instance? If so, then the log calls should be changed to include the PIP instance name , probably...